src/sentry/static/sentry/app/views/organizationIntegrations/abstractIntegrationDetailedView.tsx

```diff
@@ -142,6 +142,46 @@ class AbstractIntegrationDetailedView<
     return null;
   }
 
+  renderEmptyConfigurations() {
+    const {organization} = this.props;
+    const {IntegrationDirectoryFeatures} = getIntegrationFeatureGate();
+
+    return (
+      <EmptyConfigurationContainer>
+        <EmptyConfigurationTitle>You haven't set anything up yet</EmptyConfigurationTitle>
+        <EmptyConfigurationBody>
+          But that doesn’t have to be the case for long! Add an installation to get
+          started.
+        </EmptyConfigurationBody>
+        <div>
+          <IntegrationDirectoryFeatures {...this.featureProps}>
```

Comment:
@NisanthanNanthakumar What's the difference between this `IntegrationDirectoryFeatures` component instance and the and the one in `renderAddInstallButton`?

Reply:
@scefali if it's disabled, it renders an empty div. Inside `this.renderAddInstallButton`, we call `this.renderTopButton` which controls the button view and is implemented by the derived classes.

Reply:
@NisanthanNanthakumar wondering if there is a way to reduce code repetition since a lot looks similar

Reply:
@scefali wrote another method called `renderButton` for `AbstractIntegrationDetailedView` that does the conditional logic and calls `renderTopButton`. `this.renderAddInstallButton` now calls `renderButton`.

Reply:
@NisanthanNanthakumar Hmm...ok. It still looks like the following code is repeated twice:
```
                <DisableWrapper>
                  <Access organization={organization} access={['org:integrations']}>
                    {({hasAccess}) => (
                      <Tooltip
                        title={t(
                          'You must be an organization owner, manager or admin to install this.'
                        )}
                        disabled={hasAccess}
                      >
                        {this.renderTopButton(disabled, hasAccess)}
                      </Tooltip>
                    )}
                  </Access>
                  {disabled && <DisabledNotice reason={disabledReason} />}
                </DisableWrapper>
```
Your call if there is a way to encanpsulate this in a way that isn't repeating itself

Reply:
@scefali sorry wrote that comment before I pushed the new updates